internal/types/path.go

```diff
@@ -17,17 +17,81 @@ package types
 import (
 	"fmt"
 	"strconv"
+	"strings"
 )
 
+// Path represents field path type. It should be used wherever we work with paths or dot notation.
+type Path struct {
+	s []string
+}
+
+// NewPath returns Path from a strings slice. Provided strings slice would be copied into a new Path.
+func NewPath(path ...string) Path {
+	p := Path{s: make([]string, len(path))}
+	copy(p.s, path)
+	return p
+}
+
+// NewPathFromString returns Path from path string. Path string should contain fields separated with '.'.
+func NewPathFromString(s string) Path {
+	path := strings.Split(s, ".")
+
+	return NewPath(path...)
+}
+
+// Len returns path length.
+func (p Path) Len() int {
+	return len(p.s)
+}
+
+// Slice returns path slice copy.
+func (p Path) Slice() []string {
+	path := make([]string, len(p.s))
+	copy(path, p.s)
+	return path
+}
+
+// Suffix returns the first path element.
+func (p Path) Suffix() string {
```

Comment:
Actually, let's keep those names, but fix descriptions

```suggestion
// Suffix returns the last path element.
func (p Path) Suffix() string {
```